Extension/src/LanguageServer/configurations.ts

```diff
@@ -883,6 +890,9 @@ export class CppProperties {
                             configuration.macFrameworkPath = this.defaultFrameworks;
                         }
                     }
+                } else {
+                    // add compiler to list of trusted compilers
+                    util.addTrustedCompiler(configuration.compilerPath);
```

Comment:
It looks like it's adding null, and 'Select another compiler on my machine' at this point. The 'Select another compiler on my machine' case happens if the user hits Escape on the open file dialog box when selecting the compiler.